proof/crefine/AARCH64/Fastpath_C.thy

```diff
@@ -1736,776 +1684,988 @@ lemma recv_ep_queued_st_tcb_at':
     apply (drule pspace_alignedD')
     apply simp+
-    apply (simp add:projectKO_opt_tcb objBitsKO_def
-      split: Structures_H.kernel_object.splits)
+    apply (simp add: projectKO_opt_tcb objBitsKO_def
+                split: Structures_H.kernel_object.splits)
     done
 
-  (* FIXME indentation is wonky in this proof, fix will come in a future patch, hopefully when
-     automatic indentation is improved *)
   show ?thesis
-  supply if_cong[cong] option.case_cong[cong]
-  apply (cinit lift: cptr_' msgInfo_')
+    supply if_cong[cong] option.case_cong[cong] Collect_const[simp del]
+    apply (cinit lift: cptr_' msgInfo_')
+     (* this also lifts out pickFastpath alternative to general alternative, but not clear what
+        pickFastpath is for  *)
      apply (simp add: catch_liftE_bindE unlessE_throw_catch_If
                       unifyFailure_catch_If catch_liftE
                       getMessageInfo_def alternative_bind
-                cong: if_cong call_ignore_cong del: Collect_const)
+                cong: if_cong call_ignore_cong)
      apply (rule ccorres_pre_getCurThread)
      apply (rename_tac curThread)
      apply (rule ccorres_symb_exec_l3[OF _ user_getreg_inv' _ empty_fail_user_getreg])+
        apply (rename_tac msginfo' cptr')
        apply (rule_tac P="msginfo' = msginfo \<and> cptr' = cptr" in ccorres_gen_asm)
-       apply (simp del: Collect_const cong: call_ignore_cong)
-       apply (simp only: )
+       (* the call_ignore_cong in this proof is required to prevent corruption of arguments in
+          endpoint_ptr_mset_epQueue_tail_state_'proc so that eventually fastpath_dequeue_ccorres
+          can apply  *)
+       apply (simp cong: call_ignore_cong)
+       apply (simp only:)
        apply (csymbr, csymbr)
-       apply (rule ccorres_rhs_assoc2)
-       apply (rule_tac r'="\<lambda>ft ft'. (ft' = scast seL4_Fault_NullFault) = (ft = None)"
-                 and xf'="ret__unsigned_'" in ccorres_split_nothrow)
+       (* get fault type *)
+       apply csymbr
+       apply (rule_tac r'="\<lambda>ft ft'. (ft' = scast seL4_Fault_NullFault) = (ft = None)" and
+                       xf'=ret__unsigned_longlong_' in ccorres_split_nothrow)
            apply (rule_tac P="cur_tcb' and (\<lambda>s. curThread = ksCurThread s)"
-                     in ccorres_from_vcg[where P'=UNIV])
+                           in ccorres_from_vcg[where P'=UNIV])
            apply (rule allI, rule conseqPre, vcg)
            apply (clarsimp simp: cur_tcb'_def rf_sr_ksCurThread)
            apply (drule(1) obj_at_cslift_tcb, clarsimp)
            apply (clarsimp simp: typ_heap_simps' ctcb_relation_def cfault_rel_def)
            apply (rule rev_bexI, erule threadGet_eq)
            apply (clarsimp simp: seL4_Fault_lift_def Let_def split: if_split_asm)
           apply ceqv
+         apply (rename_tac fault fault')
+         apply (rule ccorres_alternative1) (* pick pickFastpath = True, still not clear what it's for *)
+
```

Comment:
Another newline here, maybe unintentional

Reply:
yeah, better without it